### PR TITLE
Added sin to tensorflow frontend

### LINKED ISSUE #12916: sin

### PR DESCRIPTION
**Pull request by vtsamit**
Close  #12916

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -325,6 +325,15 @@ def squared_difference(x, y, name=None):
     return ivy.square(ivy.subtract(x, y))
 
 
+@with_supported_dtypes(
+    {"2.9.0 and below": ("float16", "float32", "float64", "complex64", "complex128")},
+    "tensorflow",
+)
```
Comment (hmahmood24) [inline ivy/functional/frontends/tensorflow/math.py:331]: `tensorflow` actually does support `bfloat16` in inputs. You can verify that [here](https://www.tensorflow.org/api_docs/python/tf/math/sin). Maybe you can add that here 🙂

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @vtsamit, Thanks for a great contribution! I've requested a change. Kindly look into that.

There is also one more thing. There is already an implementation of `sin` in `ivy/functional/frontends/tensorflow/raw_ops.py`. In such scenarios, you should remove that implementation from `ivy/functional/frontends/tensorflow/raw_ops.py` and only keep an alias to your current implementation there as mentioned [here](https://github.com/unifyai/ivy/issues/1545#issue-1276763888) and [here](https://github.com/unifyai/ivy/issues/1565#issue-1277292523). Kindly look into this and remove the existing implmenetation of `sin` from `ivy/functional/frontends/tensorflow/raw_ops.py` and only keep an alias over there. 😊

File: ivy/functional/frontends/tensorflow/math.py (hunk 1 above)
Comment (vtsamit) [reply ivy/functional/frontends/tensorflow/math.py]: Thank you sir. I have made the desired changes.

**Comment by hmahmood24** [comment]
Thanks a lot @vtsamit! There are just a few more things I noticed. 

There seem to be some linting errors with your PR as diagnosed [here](https://github.com/unifyai/ivy/actions/runs/4520504930/jobs/7961659989?pr=13106). They should be fairly easy to fix. I'm assuming you haven't installed `pre-commit` because it automatically runs these linting and formatting checks everytime you push your code. It is essential that you set it up since we rigorously follow it internally everywhere in our codebase. You can find the instructions to install it and set it up [here](https://lets-unify.ai/ivy/overview/contributing/setting_up.html#pre-commit). Make sure to fix any errors it picks up because our CI tests are going to fail otherwise.

Also, can you please manually run the test you added in your local environment and paste a screenshot of the results here? The CI doesn't seem to be able to collect and run your tests for some reason.

**Review by hmahmood24 — APPROVED** [review]

**Comment by vtsamit** [comment]
Screenshot from 2023-03-27 01-04-44

**Comment by vtsamit** [comment]
Sir, these are the results of pre-commit tests. It automatically made the desired changes for linting errors. Thank you for making me aware of the pre-commit tests.

**Comment by hmahmood24** [comment]
Hey @vtsamit! Thank you for fixing the linting errors. They seem to pass the checks now. As for the tests, I meant could you please run your function's test using `pytest` and show me the results for that? Thanks! 🙂

**Comment by vtsamit** [comment]
Screenshot from 2023-04-03 23-18-26

**Comment by vtsamit** [comment]
Screenshot from 2023-04-03 23-18-43

**Comment by hmahmood24** [comment]
Hey @vtsamit, the packages in your environment seem to be broken. Kindly re-install the packages from the `requirements/requirements.txt` and `requirements/optional.txt` to set up your environment again or rebuild your container (if working in a devcontainer) to make sure these tests run fine

**Comment by vtsamit** [comment]
Okay sir. I am re-installing the required packages. After this, I hope the tests would run fine. Thank you.

**Comment by vtsamit** [comment]
Sir, even after reinstalling all the packages and building a new conda environment, the same error is getting displayed ( the images attached above) after running the pytest for sin function in test_math.py. The error report is showing AttributeError: 'NoneType' object has no attribute 'backend'.

**Comment by vtsamit** [comment]
Sir, now the "intelligent-tests-pr" is also successful.

**Pull request merged**